src/main/java/frc/trigon/robot/subsystems/swerve/SwerveConstants.java

```diff
@@ -39,30 +40,18 @@ public class SwerveConstants {
             REAR_LEFT_WHEEL_DIAMETER_METERS = RobotHardwareStats.isSimulation() ? 0.1016 : 0.049274 * 2,
             REAR_RIGHT_WHEEL_DIAMETER_METERS = RobotHardwareStats.isSimulation() ? 0.1016 : 0.049274 * 2;
     static final SwerveModule[] SWERVE_MODULES = {
-            new SwerveModule(FRONT_LEFT_ID, FRONT_LEFT_STEER_ENCODER_OFFSET, FRONT_LEFT_WHEEL_DIAMETER_METERS),
-            new SwerveModule(FRONT_RIGHT_ID, FRONT_RIGHT_STEER_ENCODER_OFFSET, FRONT_RIGHT_WHEEL_DIAMETER_METERS),
-            new SwerveModule(REAR_LEFT_ID, REAR_LEFT_STEER_ENCODER_OFFSET, REAR_LEFT_WHEEL_DIAMETER_METERS),
-            new SwerveModule(REAR_RIGHT_ID, REAR_RIGHT_STEER_ENCODER_OFFSET, REAR_RIGHT_WHEEL_DIAMETER_METERS)
+            new SwerveModule(FRONT_LEFT_ID, FRONT_LEFT_WHEEL_DIAMETER_METERS, FRONT_LEFT_STEER_ENCODER_OFFSET),
+            new SwerveModule(FRONT_RIGHT_ID, FRONT_RIGHT_WHEEL_DIAMETER_METERS, FRONT_RIGHT_STEER_ENCODER_OFFSET),
+            new SwerveModule(REAR_LEFT_ID, REAR_LEFT_WHEEL_DIAMETER_METERS, REAR_LEFT_STEER_ENCODER_OFFSET),
+            new SwerveModule(REAR_RIGHT_ID, REAR_RIGHT_WHEEL_DIAMETER_METERS, REAR_RIGHT_STEER_ENCODER_OFFSET)
     };
 
     private static final DoubleSupplier SIMULATION_YAW_VELOCITY_SUPPLIER = () -> RobotContainer.SWERVE.getSelfRelativeVelocity().omegaRadiansPerSecond;
-
-    private static final double
-            FRONT_MODULE_X_DISTANCE_FROM_CENTER = 0.25,
-            FRONT_MODULE_Y_DISTANCE_FROM_CENTER = 0.25,
-            REAR_MODULE_X_DISTANCE_FROM_CENTER = 0.25,
-            REAR_MODULE_Y_DISTANCE_FROM_CENTER = 0.25;
-    public static final Translation2d[] MODULE_LOCATIONS = {
-            new Translation2d(FRONT_MODULE_X_DISTANCE_FROM_CENTER, FRONT_MODULE_Y_DISTANCE_FROM_CENTER),
-            new Translation2d(FRONT_MODULE_X_DISTANCE_FROM_CENTER, -FRONT_MODULE_Y_DISTANCE_FROM_CENTER),
-            new Translation2d(REAR_MODULE_X_DISTANCE_FROM_CENTER, REAR_MODULE_Y_DISTANCE_FROM_CENTER),
-            new Translation2d(REAR_MODULE_X_DISTANCE_FROM_CENTER, -REAR_MODULE_Y_DISTANCE_FROM_CENTER)
-    };
-    public static final SwerveDriveKinematics KINEMATICS = new SwerveDriveKinematics(MODULE_LOCATIONS);
+    public static final SwerveDriveKinematics KINEMATICS = new SwerveDriveKinematics(PathPlannerConstants.getRobotConfig().moduleLocations);
     private static final double FURTHEST_MODULE_DISTANCE_FROM_CENTER = Math.hypot(
-            REAR_MODULE_X_DISTANCE_FROM_CENTER, FRONT_MODULE_Y_DISTANCE_FROM_CENTER
+            PathPlannerConstants.getRobotConfig().moduleLocations[0].getX(),
+            PathPlannerConstants.getRobotConfig().moduleLocations[3].getY()
```

Comment:
huh?

Reply:
what? This is the same logic as there was earlier

Reply:
why the third module

Reply:
It's the 4th. Arrays start from 0.

Reply:
Doesn't change my question much
Why